src/main/com/example011/radek/v1/Mines.java

```diff
@@ -0,0 +1,146 @@
+package main.com.example011.radek.v1;
+
+import java.security.NoSuchAlgorithmException;
+import java.security.SecureRandom;
+import java.util.Scanner;
+
+public class Mines {
+
+    protected int playgroundSize;
+
+    protected int numMines;
+
+    protected String[][] playground;
+
+    public Mines(int size, int mines) {
+
+        if (setPlaygroundSize(size)) {
+            setNumMines(mines);
+        } else {
+            System.out.println("Bad size of playground, size must be bigger then 5!");
+            System.exit(0);
+        }
+    }
+
+    public void fillPlaygroundWithMines() throws NoSuchAlgorithmException {
+
+        playground = new String[playgroundSize][playgroundSize];
+        SecureRandom random = SecureRandom.getInstanceStrong();
+        int placedMines = 0;
+        while (placedMines != getNumMines() || placedMines == 0) {
+            int generatedRowNumber = random.nextInt(playgroundSize);
+            int generatedColumnNumber = random.nextInt(playgroundSize);
+            if (playground[generatedRowNumber][generatedColumnNumber] == null) {
+                playground[generatedRowNumber][generatedColumnNumber] = "x";
+                placedMines = placedMines + 1;
+            }
+        }
+    }
+
+    public void printPlayground(boolean showMines) {
```

Comment:
Tuto metodu přepracovat. Strašně moc velká. Pojdme se dostat aspon na poloviční počet řádk§ pro začátek.

Reply:
Z 27 na 13 řádků, další optimalizaci bych viděl jedině při přepsání z 2 for cycklů na list, listů a jejich následné zpracování, ale ještě je neumím dostatečně, abych to poskládal, nepodařilo se mi to zatím

Reply:
ok, nechme to uz